app/models/form/custom_emoji_batch.rb

```diff
@@ -49,54 +49,48 @@ def update!
   end
 
   def list!
-    custom_emojis.each { |custom_emoji| authorize(custom_emoji, :update?) }
-
     custom_emojis.each do |custom_emoji|
+      authorize(custom_emoji, :update?)
```

Comment:
iirc loops like this are explicitly split in two loops to avoid having some emoji updated and the remaining not updated because of a failed authorization. I guess the same could be achieved with a transaction but it's a slightly heavier mechanism.

Not that it matters much in practice as the ability of a user to update an emoji is currently not dependent on the emoji, only the user.